common/src/cookie/cookie.service.ts

```diff
@@ -0,0 +1,142 @@
+import { DOCUMENT, isPlatformBrowser } from '@angular/common';
+import { Injectable, InjectFlags, Injector, PLATFORM_ID } from '@angular/core';
+import { Request, Response } from 'express';
+import { isEmpty } from 'lodash';
+import { CookieOptions } from './models';
+import { REQUEST, RESPONSE } from './tokens';
+
+@Injectable()
+export class CookieService<TKey extends string = string> {
+  protected document: Document;
+  protected platformID: object;
+  protected request: Request;
+  protected response: Response;
+  protected defaultOptions: CookieOptions;
+  protected isDocumentAccessible: boolean;
+
+  protected get cookieString(): string {
+    return this.isDocumentAccessible ? this.document.cookie : this.request.headers.cookie;
+  }
+
+  constructor(
+    protected injector: Injector
+  ) {
+    this.document = this.injector.get(DOCUMENT);
+    this.platformID = this.injector.get(PLATFORM_ID);
+    this.request = this.injector.get(REQUEST, undefined, InjectFlags.Optional);
+    this.response = this.injector.get(RESPONSE, undefined, InjectFlags.Optional);
+    this.defaultOptions = this.injector.get(CookieOptions, undefined, InjectFlags.Optional);
+    this.isDocumentAccessible = isPlatformBrowser(this.platformID);
+  }
+
+  public check(key: TKey): boolean {

```

Comment:
@VladimirGlinskii 

Maybe we don't need this method. If it's needed, then we could rename it to `has`.